File: binary-search.c
#include <stdio.h>

int main()
{
    int listValues[] = {0, 2, 4, 6, 8, 9, 10, 12, 62, 70, 71, 87, 90, 95, 98, 100};
    const size_t listLength = (sizeof(listValues) / sizeof(listValues[0]));
    int target = 0;
    int firstOfList = 0, endOfList, middleOfList;

    endOfList = listLength - 1;
    middleOfList = (firstOfList + endOfList) / 2;

    do
    {
        if (listValues[middleOfList] < target)
        {
            firstOfList = middleOfList + 1;
        }
        else if (listValues[middleOfList] == target)
        {
            printf("What did you looking for is in the %dth block!\n", middleOfList+1);
            return 0;
        }
        else
        {
            endOfList = middleOfList - 1;
        }

        middleOfList = (firstOfList + endOfList) / 2;
    } while (firstOfList <= endOfList);

    if (firstOfList > endOfList)
    {
        printf("Error 404!\n");
    }

    return 0;
}

File: binary-search.ts
"strict mode";

const arr: number[] = [
  0,
  2,
  4,
  6,
  8,
  9,
  10,
  12,
  62,
  70,
  71,
  87,
  90,
  95,
  98,
  100,
]; //length = 16

const target: number = 0;

let first: number = 0;
let end: number = arr.length - 1;
let middle: number;

function setMiddle(): void {
  middle = Math.floor((first + end) / 2);
}
setMiddle();

while (first <= end) {
  if (target === arr[middle]) {
    console.log(`What did you looking for is in the ${middle + 1}th block!`);
    break;
  } else if (target < arr[middle]) end = middle - 1;
  else if (target > arr[middle]) first = middle + 1;

  setMiddle();
}